subsys/shell/modules/bt.c

```diff
@@ -0,0 +1,2780 @@
+
+		if (!strcmp(dup_filter, "dups")) {
+			dups = BT_HCI_LE_SCAN_FILTER_DUP_DISABLE;
+		} else if (!strcmp(dup_filter, "nodups")) {
+			dups = BT_HCI_LE_SCAN_FILTER_DUP_ENABLE;
+		} else {
+			return -EINVAL;
+		}
+	}
+
+	action = argv[1];
+	if (!strcmp(action, "on")) {
+		cmd_active_scan_on(dups);
+	} else if (!strcmp(action, "off")) {
+		cmd_scan_off();
+	} else if (!strcmp(action, "passive")) {
+		cmd_passive_scan_on(dups);
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_connect_le(int argc, char *argv[])
+{
+	int err;
+	bt_addr_le_t addr;
+	struct bt_conn *conn;
+
+	if (argc < 3) {
+		return -EINVAL;
+	}
+
+	err = str2bt_addr_le(argv[1], argv[2], &addr);
+	if (err) {
+		printk("Invalid peer address (err %d)\n", err);
+		return 0;
+	}
+
+	conn = bt_conn_create_le(&addr, BT_LE_CONN_PARAM_DEFAULT);
+
+	if (!conn) {
+		printk("Connection failed\n");
+	} else {
+
+		printk("Connection pending\n");
+
+		/* unref connection obj in advance as app user */
+		bt_conn_unref(conn);
+	}
+
+	return 0;
+}
+
+static int cmd_disconnect(int argc, char *argv[])
+{
+	struct bt_conn *conn;
+	int err;
+
+	if (default_conn && argc < 3) {
+		conn = bt_conn_ref(default_conn);
+	} else {
+		bt_addr_le_t addr;
+
+		if (argc < 3) {
+			return -EINVAL;
+		}
+
+		err = str2bt_addr_le(argv[1], argv[2], &addr);
+		if (err) {
+			printk("Invalid peer address (err %d)\n", err);
+			return 0;
+		}
+
+		conn = bt_conn_lookup_addr_le(&addr);
+	}
+
+	if (!conn) {
+		printk("Not connected\n");
+		return 0;
+	}
+
+	err = bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	if (err) {
+		printk("Disconnection failed (err %d)\n", err);
+	}
+
+	bt_conn_unref(conn);
+
+	return 0;
+}
+
+static int cmd_auto_conn(int argc, char *argv[])
+{
+	bt_addr_le_t addr;
+	int err;
+
+	if (argc < 3) {
+		return -EINVAL;
+	}
+
+	err = str2bt_addr_le(argv[1], argv[2], &addr);
+	if (err) {
+		printk("Invalid peer address (err %d)\n", err);
+		return 0;
+	}
+
+	if (argc < 4) {
+		bt_le_set_auto_conn(&addr, BT_LE_CONN_PARAM_DEFAULT);
+	} else if (!strcmp(argv[3], "on")) {
+		bt_le_set_auto_conn(&addr, BT_LE_CONN_PARAM_DEFAULT);
+	} else if (!strcmp(argv[3], "off")) {
+		bt_le_set_auto_conn(&addr, NULL);
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_select(int argc, char *argv[])
+{
+	struct bt_conn *conn;
+	bt_addr_le_t addr;
+	int err;
+
+	if (argc < 3) {
+		return -EINVAL;
+	}
+
+	err = str2bt_addr_le(argv[1], argv[2], &addr);
+	if (err) {
+		printk("Invalid peer address (err %d)\n", err);
+		return 0;
+	}
+
+	conn = bt_conn_lookup_addr_le(&addr);
+	if (!conn) {
+		printk("No matching connection found\n");
+		return 0;
+	}
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+	}
+
+	default_conn = conn;
+
+	return 0;
+}
+
+static int cmd_conn_update(int argc, char *argv[])
+{
+	struct bt_le_conn_param param;
+	int err;
+
+	if (argc < 5) {
+		return -EINVAL;
+	}
+
+	param.interval_min = strtoul(argv[1], NULL, 16);
```

Comment:
Will any invalid input passed to `strtoul()` cause any issue? If I pass "potato" instead of a number, it'll return `ULONG_MAX`, and truncate to `typeof(interval_min)`... is this going to pose an issue?